### PR TITLE
Auto provision image stream

### PR DESCRIPTION
**Pull request by ncdc**
Auto provision an image stream when an image is pushed if the stream doesn't yet exist.

### CODE REVIEW

**Comment by ncdc** [comment]
@smarterclayton interested in doing this?

**Comment by smarterclayton** [comment]
Yes if user has permission.

**Comment by ncdc** [comment]
If a project/namespace doesn't exist yet, should this create it (assuming the user has permissions to do so)? I'm thinking no...

**Comment by smarterclayton** [comment]
No

**Comment by smarterclayton** [comment]
What's the status of this?

**Comment by ncdc** [comment]
I'll have it in once I finish reviewing Ravi's v2 auth work. That's a prerequisite.

**Comment by ncdc** [comment]
@smarterclayton @liggitt I'd like to add a test for this to e2e, but I need an oauth token and I'm not sure how to automate requesting one, as `openshift ex tokens request-token` doesn't allow you to set the username and password with flags. Is there a way to do it?

**Comment by ncdc** [comment]
[test]

**Comment by liggitt** [comment]
can you describe what you would do with the token? do you need a token or just a valid login to run osc commands with? would `osc login ...` work?

**Comment by ncdc** [comment]
I need a token so I can pass it to docker login as the password

**Comment by ncdc** [comment]
Actually I can just do this in an integration test instead

**Comment by ncdc** [comment]
@smarterclayton PTAL. This is obviously v2 registry only.

File: pkg/dockerregistry/middleware/repository/openshift.go [hunk 1]
```diff
@@ -188,9 +188,42 @@ func (r *repository) Put(ctx context.Context, manifest *manifest.SignedManifest)
 		},
 	}
 
-	if err := r.registryClient.ImageStreamMappings(r.namespace).Create(&irm); err != nil {
-		log.Errorf("Error creating ImageStreamMapping: %s", err)
-		return err
+	if err := r.registryClient.ImageStreamMappings(r.namespace).Create(&ism); err != nil {
```
Comment (smarterclayton) [inline pkg/dockerregistry/middleware/repository/openshift.go]: nit: in the future you might want to make the if block return and the else be the rest of the page (with a good comment about autoprovision).

**Comment by ncdc** [comment]
@smarterclayton any other comments or is it ok to merge when green?

**Comment by smarterclayton** [comment]
Merge when green

**Comment by ncdc** [comment]
[merge]

**Pull request merged**